### PR TITLE
Function to load reference samples from google earth engine

### PR DESCRIPTION
**Pull request by benwulfe**
We are now auto-publishing all reference isotopes to earth engine and this function loads the data into a dataframe for further manipulation (data validation, partitioning, etc).

@erickzul How did you create the canonical data sets for training, validation and test?  I assume this is a variation of partition(x).  Once I have that I can update XGB to grab EE, partition and create the isoscape.

### CODE REVIEW

File: dataset.py [hunk 1]
```diff
@@ -254,6 +259,33 @@ def partitioned_reference_data(reference_csv_filename: str) -> PartitionedDatase
...
+  for filter_fc in filters:
+    fc = fc.filter(filter_fc)
+  info = fc.getInfo()
+  features = info['features']
+  dictarr = []
+
+  for f in features:
+      attr = f['properties']
+      attr[_LATITUDE_COLUMN_NAME] = f['geometry']['coordinates'][1]
+      attr[_LONGITUDE_COLUMN_NAME] = f['geometry']['coordinates'][0]
```
Comment (rothn) [inline dataset.py:285]: Maybe a dumb questions, but are we sure it's lon/lat not lat/lon?

**Review by rothn — CHANGES_REQUESTED** [review]
Looks good in general! See my comments.

File: gee_datasource_test.ipynb
Comment (rothn) [inline gee_datasource_test.ipynb:1]: > "Test the API by printing the elevation of Mount Everest."

Is this still what `load_reference_samples()` does if called without arguments?

**Comment by benwulfe** [comment]
> In gee_datasource_test.ipynb, where is the 1000 number coming from?

That is the current count from earth engine. I am assuming that number should only increase.  I did not want to verify any specific contents because that would leak information.

File: gee_datasource_test.ipynb
Comment (benwulfe) [reply gee_datasource_test.ipynb:1]: no, that's just a simple test of the earth engine API.  I want to leave that in there, but will update the comment.  It's actually from the EE colab example.

File: dataset.py (hunk 1 above)
Comment (benwulfe) [reply dataset.py:285]: yes, I double checked.  I will triple check, by comparing the reference data with what we expect.

Comment (benwulfe) [reply dataset.py:285]: triple checked and I added some minimal output that shows 5 example lat/longs as well as another assert that ensure min(long) < -70.

File: gee_datasource_test.ipynb
Comment (benwulfe) [reply gee_datasource_test.ipynb:1]: i actually simply removed this since I had previously removed the code its talking about

**Review by rothn — APPROVED** [review]

**Pull request merged**